.github/workflows/lint-swiftlint-formatting.yml

```diff
@@ -51,3 +48,24 @@ jobs:
       - name: Run CI Diagnostics
         if: failure()
         run: ./scripts/ci-diagnostics.sh
+
+  # This check validates that either lint passed or was skipped, which allows us
+  # to make lint-swiftlint a required check with only running the lint when required.
+  # So, we don't have to run lint-swiftlint, for example, for unrelated changes.
+  lint_swiftlint-required-check:
+    needs:
+      [
+        files-changed,
+        lint,
+      ]
+    name: Lint SwiftLint
+    # This is necessary since a failed/skipped dependent job would cause this job to be skipped
+    if: always()
+    runs-on: ubuntu-latest
+    steps:
+      # If any jobs we depend on fails gets cancelled or times out, this job will fail.
+      # Skipped jobs are not considered failures.
+      - name: Check for failures
+        if: contains(needs.*.result, 'failure') || contains(needs.*.result, 'cancelled')
+        run: |
+          echo "One of the lint-swiftlint jobs has failed." && exit 1
```

Comment:
`m`: The workflow file only contains one job. It feels a bit of an overkill to me to add a required check for this. The check is more code, more complexity. This applies to all PRs.

Reply:
We decided a couple of months ago internally that we require formatting and linting to pass for PRs. I can not enable GitHub's "Automerge" without a required status check for this behaviour. Either we keep keep the file change detection or run the linting at all times. We can evaluate if merging the linting workflow has benefits, but each one has different file change paths, so we would have to build a more complex workflow logic (which is why I separated it in the first place during the workflow name refactoring)

Reply:
Ah now I understand. In this case I agree with it being an additional job, but I'll keep it this way for now for consistency.

Reply:
Yes, consistency is an argument in its favor. Up to you.